pkg/cgroups/cgroups.go

```diff
@@ -128,7 +128,25 @@ func init() {
 // getAvailableControllers get the available controllers
 func getAvailableControllers(exclude map[string]controllerHandler, cgroup2 bool) ([]controller, error) {
 	if cgroup2 {
-		return nil, fmt.Errorf("getAvailableControllers not implemented yet for cgroup v2")
+		controllers := []controller{}
+		// Get caller uid to confirm if its invoked by root or non-root
+		uid := os.Geteuid()
+		subtreeControl := cgroupRoot + "/cgroup.subtree_control"
+		if uid != 0 {
+			subtreeControl = fmt.Sprintf("%s/user.slice/user-%d.slice/user@%d.service/cgroup.subtree_control", cgroupRoot, uid, uid)
```

Comment:
Not robust

https://github.com/kubernetes-sigs/kind/issues/2254#issuecomment-842888408

Reply:
@AkihiroSuda ah I think i'll get prefix from group manager, that should resolve this. Thanks for the review :) i'll try to fix these.

Reply:
can't we read those from the current cgroup so we don't have to worry about systemd/cgroupfs or rootless/rootful?

Reply:
probably for root it still makes sense to read them from the root cgroup

Reply:
@giuseppe Yeah I think for rootless i could just get available controllers from current cgroup. That should be more standard.

Reply:
@AkihiroSuda @giuseppe PTAL for `rootless cgroupv2` now we are marking the controller availability as `false` if it got failed during `initialization` of cgroup. This seems to be more simple and wouldn't require handling caveats of systemd/cgroupfs unit name.

Reply:
@AkihiroSuda @giuseppe PTAL made some changes I think current checking available controller for current user is better. `Service` and `scope` would anyways inherit this. Here is the output for `rootless` / `cgroupv2` from my machine.

```bash
host:
  arch: amd64
  buildahVersion: 1.20.2-dev
  cgroupInfo:
    cpuShares: false
    memoryLimit: true
    pidsLimit: true
  cgroupManager: systemd
  cgroupVersion: v2
```

Reply:
@giuseppe @AkihiroSuda This is resolved in latest commit.